blockstore/src/main/java/brooklyn/location/blockstore/ec2/Ec2VolumeCustomizers.java

```diff
@@ -69,104 +70,172 @@ public static JcloudsLocationCustomizer withNewVolumes(List<Integer> capacities)
+                ((EC2TemplateOptions) templateOptions).mapNewVolumeToDeviceName(
+                        ebsVolumeManager.getVolumeDeviceName(blockOptions.getDeviceSuffix()), blockOptions.getSizeInGb().intValue(), blockOptions.deleteOnTermination());
             }
+        }
 
-            @Override
-            public void customize(JcloudsLocation location, ComputeService computeService, JcloudsMachineLocation machine) {
-                for (Map.Entry<BlockDeviceOptions, FilesystemOptions> entry : volumes.entrySet()) {
-                    BlockDeviceOptions blockOptions = entry.getKey();
-                    FilesystemOptions filesystemOptions = entry.getValue();
-                    if (filesystemOptions != null) {
-                    	BlockDeviceOptions blockOptionsCopy = BlockDeviceOptions.copy(blockOptions);
-                    	Optional<NodeMetadata> node = machine.getOptionalNode();
-                    	if (node.isPresent()) {
-                    		blockOptionsCopy.zone(node.get().getLocation().getId());
-                    	}
-                        ebsVolumeManager.createAttachAndMountVolume(machine, blockOptionsCopy, filesystemOptions);
+        @Override
+        public void customize(JcloudsLocation location, ComputeService computeService, JcloudsMachineLocation machine) {
+            for (Map.Entry<BlockDeviceOptions, FilesystemOptions> entry : volumes.entrySet()) {
+                BlockDeviceOptions blockOptions = entry.getKey();
+                FilesystemOptions filesystemOptions = entry.getValue();
+                if (filesystemOptions != null) {
+                    BlockDeviceOptions blockOptionsCopy = BlockDeviceOptions.copy(blockOptions);
+                    Optional<NodeMetadata> node = machine.getOptionalNode();
+                    if (node.isPresent()) {
+                        blockOptionsCopy.zone(node.get().getLocation().getId());
                     }
+                    mountedBlockDevice = ebsVolumeManager.createAttachAndMountVolume(machine, blockOptionsCopy, filesystemOptions);
                 }
             }
-        };
+        }
     }
 
     // TODO: Either the JavaDoc or the implementation is incorrect. The implementation makes no attempt to attach volumes.
     /**
-     * Returns a location customizer that:
+     * A location customizer type that:
      * <ul>
      * <li>configures the EC2 availability zone</li>
      * <li>obtains a new EBS volume from the specified snapshot in the given availability zone</li>
      * <li>attaches the new volume to the newly-provisioned EC2 instance</li>
      * <li>mounts the filesystem under the requested path</li>
      * </ul>
      */
-    public static JcloudsLocationCustomizer withExistingSnapshot(final AttachedBlockDevice attachedDevice,
-            final BlockDeviceOptions blockOptions, final FilesystemOptions filesystemOptions) {
 
-        return new BasicJcloudsLocationCustomizer() {
```

Comment:
As this is new functionality, not used in the wild, then can refactor this without keeping backwards compatibility.